exchanges/coinbene/coinbene_websocket.go

```diff
@@ -0,0 +1,321 @@
+			if err != nil {
+				c.Websocket.DataHandler <- err
+				return
+			}
+			c.Websocket.TrafficAlert <- struct{}{}
+			if string(stream.Raw) == "ping" {
+				c.WebsocketConn.Lock()
+				c.WebsocketConn.Connection.WriteMessage(websocket.TextMessage, []byte("pong"))
+				c.WebsocketConn.Unlock()
+				continue
+			}
+			var result map[string]interface{}
+			err = common.JSONDecode(stream.Raw, &result)
+			if err != nil {
+				c.Websocket.DataHandler <- err
+			}
+			_, ok := result["event"]
+			switch {
+			case ok && result["event"].(string) == "subscribe":
+				continue
+			case ok && result["event"].(string) == "unsubscribe":
+				continue
+			case ok && result["event"].(string) == "error":
+				c.Websocket.DataHandler <- fmt.Errorf("message: %s. code: %v", result["message"], result["code"])
+				continue
+			}
+			if ok && strings.Contains(result["event"].(string), "login") {
+				if result["success"].(bool) {
+					c.GenerateAuthSubs()
+				}
+				continue
+			}
+			switch {
+			case strings.Contains(result["topic"].(string), "ticker"):
+				var ticker WsTicker
+				err = common.JSONDecode(stream.Raw, &ticker)
+				if err != nil {
+					c.Websocket.DataHandler <- err
+					continue
+				}
+				for x := range ticker.Data {
+					c.Websocket.DataHandler <- wshandler.TickerData{
+						Quantity:   ticker.Data[x].Volume24h,
+						ClosePrice: ticker.Data[x].LastPrice,
+						HighPrice:  ticker.Data[x].High24h,
+						LowPrice:   ticker.Data[x].Low24h,
+						Pair:       currency.NewPairFromString(ticker.Data[x].Symbol),
+						Exchange:   c.Name,
+						AssetType:  orderbook.Swap,
+					}
+				}
+			case strings.Contains(result["topic"].(string), "tradeList"):
+				var tradeList WsTradeList
+				err = common.JSONDecode(stream.Raw, &tradeList)
+				if err != nil {
+					c.Websocket.DataHandler <- err
+					continue
+				}
+				time, err1 := time.Parse(time.RFC3339, tradeList.Data[0][3])
+				if err1 != nil {
+					c.Websocket.DataHandler <- err1
+					continue
+				}
+				price, err2 := strconv.ParseFloat(tradeList.Data[0][0], 64)
+				if err2 != nil {
+					c.Websocket.DataHandler <- err2
+					continue
+				}
+				amount, err3 := strconv.ParseFloat(tradeList.Data[0][2], 64)
+				if err3 != nil {
+					c.Websocket.DataHandler <- err3
+					continue
+				}
+				c.Websocket.DataHandler <- wshandler.TradeData{
+					CurrencyPair: currency.NewPairFromString(strings.Replace(tradeList.Topic, "tradeList.", "", 1)),
+					Timestamp:    time,
+					Price:        price,
+					Amount:       amount,
+					Exchange:     c.Name,
+					AssetType:    orderbook.Swap,
+					Side:         tradeList.Data[0][1],
+				}
+			case strings.Contains(result["topic"].(string), "orderBook"):
+				var orderBook WsOrderbook
+				err = common.JSONDecode(stream.Raw, &orderBook)
+				if err != nil {
+					c.Websocket.DataHandler <- err
+					continue
+				}
+				var asks, bids []orderbook.Item
+				for i := range orderBook.Data[0].Asks {
+					var tempAsks orderbook.Item
```

Comment:
You can drop this intermidate var and do the instantiation in append(). Shazbert has just cleaned up a bunch of these in engine so it is nice to keep this consistent